packages/mui-lab/src/Alert/Alert.js

```diff
@@ -4,10 +4,7 @@ import Alert from '@mui/material/Alert';
 
 let warnedOnce = false;
 
-/**
- * @ignore - do not document.
- */
-export default React.forwardRef(function DeprecatedAlert(props, ref) {
+const warn = () => {
```

Comment:
Which was the error being reported by eslint-plugin-react-compiler?

Reply:
>Unexpected reassignment of a variable which was defined outside of the component. Components and hooks should be pure and side-effect free, but variable reassignment is a form of side-effect. If this variable is used in rendering, use useState instead. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#side-effects-must-run-outside-of-render)eslint(react-compiler/react-compiler)

reason for error is `warnedOnce` is defined outside of component and changing inside of component